src/pages/index.js

```diff
@@ -100,15 +86,356 @@ const cardSection = new Section(
+      
+        profileFormPopup.close();
+      }, 500);
+    });
+  }
+  
+  function handleAddCardFormSubmit(formData) {
+    const cardData = {
+      title: formData.title,
+      url: formData.url
+    };
+    const addCardSubmitButton = document.querySelector("#add-card-submit-button");
+    updateButtonState(addCardSubmitButton, true);
+
+    api.addCard(cardData)
+      .then((savedCardData) => {
+        console.log("Card saved to server:", savedCardData);
+
+        if (savedCardData && savedCardData._id) {
+          const card = createCard({
+            name: savedCardData.name,
+            link: savedCardData.link,
+            _id: savedCardData._id
+          });
+          cardSection.addItem(card);
+        } else {
+          console.error("Card data missing _id:", savedCardData);
+        }
+      })
+      .catch((error) => {
+        console.error("Error adding card:", error);
+      })
+      .finally(() => {
+        setTimeout(() => {
+        updateButtonState(addCardSubmitButton, false, "Create");
+        
+          addCardFormPopup.close();
+        }, 500);
+      });
+}
+  
+  function handleDeleteCard(card) {
+    if (card._id) {
+      console.log("Deleting card with ID:", card._id);
+      deleteConfirm.setSubmitFunction(() => {
+        api.deleteCard(card._id)
+          .then(() => {
+            console.log("Card deleted successfully:", card._id);
+            card.handleDeleteCard();
+          })
+          .catch((error) => {
+            console.error("Error Deleting Card:", error);
+          });
+      });
+      deleteConfirm.open();
+    } else {
+      console.error("Card ID is missing:", card);
+    }
+  }
+        
+  function handleAvatarFormSubmit(formData) {
+    const avatarFormSubmitButton = document.querySelector("#avatar-submit-button");
+    updateButtonState(avatarFormSubmitButton, true);
+    const avatarUrl = formData.avatarUrl;
+
+    api.updateUserAvatar(avatarUrl)
+        .then((userData) => {
+            console.log("API Response for Avatar Update:", userData);
+            
+            if (userData && userData.avatar) {
+                avatarImage.src = userData.avatar;
+                console.log("Avatar updated successfully:", userData.avatar);
+            } else {
+                console.error("Avatar URL missing in response");
+            }
+        })
+        .catch((error) => {
+            console.error("Error updating avatar:", error);
+            alert("Error updating avatar. Please try again.");
+        })
+        .finally(() => {
+          setTimeout(() => {
+            updateButtonState(avatarFormSubmitButton, false);
+            
+              avatarPopup.close();
+            }, 500);
+        });
+}
+  
+function handleLikeIcon(card) {
+  console.log("Toggling like for card ID:", card._id);
+  if (card._cardEl.querySelector(".card__like-button").classList.contains("card__like-button_active")) {
```

Comment:
Please, don't search the heart in `index.js`

Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle


Please, make it work like this:

```js

 if (card.isLiked) {
```